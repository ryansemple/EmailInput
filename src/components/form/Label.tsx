import React from "react";

interface LabelProps {
	className: string,
	text: string,
	htmlFor?: string
}

const Label = (props: LabelProps) => (
	<label
		htmlFor={props.htmlFor}
		className={`form_label ${props.className}`}>
		{props.text}
	</label>
)

export default Label;