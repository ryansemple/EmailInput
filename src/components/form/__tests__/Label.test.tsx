import React from "react";
import { render } from "@testing-library/react";
import Label from "../Label";

test("Renders Label component correctly.", () => {
	const { container } = render
	(
		<Label
			className="class"
			text="Label text"
			htmlFor="input"
		/>
	);
	expect(container).toMatchSnapshot();
});