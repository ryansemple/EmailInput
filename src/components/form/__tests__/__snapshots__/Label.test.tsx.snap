// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Label component correctly. 1`] = `
<div>
  <label
    class="Label class"
    for="input"
  >
    Label text
  </label>
</div>
`;
